server/aap/commands/fulfill_image_assignments.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import logging
+import superdesk
+from superdesk.utc import utcnow, utc_to_local
+from datetime import timedelta
+from planning.common import ASSIGNMENT_WORKFLOW_STATE
+from eve.utils import ParsedRequest
+from flask import json
+from flask import current_app as app
+import requests
+from lxml import etree
+from superdesk.celery_task_utils import get_lock_id
+from superdesk.lock import lock, unlock
+
+logger = logging.getLogger(__name__)
+
+
+class FullfillImageAssignments(superdesk.Command):
+    # The headers returned by the login request to the DC Rest service
+    dc_headers = dict()
+
+    def _dc_login(self):
+        """
+        Function will log into the DC system rest API and save the response headers
+        :return:
+        """
+        url = app.config.get('DC_URL')
+        username = app.config.get('DC_USERNAME')
+        password = app.config.get('DC_PASSWORD')
+
+        response = requests.get(url + '/?login[username]={}&login[password]={}'.format(username, password))
+        response.raise_for_status()
+
+        self.dc_headers = response.headers
+
+    def _get_dc_image_by_id(self, collection, id):
+        """
+        This function given a DC collection and the DC id of an image will return the meta data from DC for that image
+        :param collection:
+        :param id:
+        :return: The XML response that should contain the metadata for the single image
+        """
+        url = app.config.get('DC_URL')
+
+        retires = 0
+        while retires < 3:
+            try:
+                response = requests.get(url + '/archives/{}?search_docs[query]=id%3d'
+                                              '{}&search_docs[format]=full'.format(collection, id),
+                                        headers={'cookie': self.dc_headers.get('Set-Cookie')})
+                response.raise_for_status()
+                break
+            except requests.exceptions.RequestException:
+                try:
```

Comment:
It might be good to log the response from the request at this point in time (regardless of authentication error or something else). This will make it easier to determine the cause of a problem if there are request errors to DC

Reply:
OK